### ISSUE TITLE
Invalid Authentication

### ISSUE THREAD

**Issue by ameran** [tuya/tuya-home-assistant#136]
Can't setup integration/add account.

I've entered all the details correctly, tried & checked several times.
Checked the SmartLife app, signed out and signed back in with the same username and password (tried several times)
I've also tried the following country codes - +27, 27, ZA. All give the same error code. 
From South Africa, closest Management site is Europe. On the IOT site, I've tried with only Europe, and also with all the other management regions selected.

On the developer site, I can can see the devices in the smart life app, I've unlinked and relinked a few times as well. 

![image](https://user-images.githubusercontent.com/2052671/125204689-c55e9980-e27e-11eb-9f70-68c281f2e005.png)

![image](https://user-images.githubusercontent.com/2052671/125204695-cd1e3e00-e27e-11eb-97cd-56040f5bd914.png)

![image](https://user-images.githubusercontent.com/2052671/125204701-d0b1c500-e27e-11eb-8991-6a50c69892c4.png)

![image](https://user-images.githubusercontent.com/2052671/125205006-5f731180-e280-11eb-85d2-58d347e44d81.png)

Home Assistant Version
2021.7.0

Tuya v2 Version
1.3.1

Logs don't show any errors related to Tuya, except this:
2021-07-11 19:35:48 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration tuya_v2 which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

Any other ideas??!?!?

**Comment by tsutsuku** [comment]
@ameran Login error code？

**Comment by ameran** [comment]
Just tried again, got this:

2021-07-12 07:17:39 INFO (SyncWorker_4) [custom_components.tuya_v2.config_flow] TuyaConfigFlow._try_login finish, response:, {'code': 1106, 'msg': 'permission deny', 'success': False, 't': 1626067059535}

**Comment by ameran** [comment]
Just tried something this user tried:
https://github.com/tuya/tuya-home-assistant/issues/90#issuecomment-874073232

Entered the gg- account name and it worked. 

Also initially created my smart life account with a Google account.

**Comment by SM-26** [comment]
I'm also getting the same problem, but my account isn't made by the google sso.

anything I could try to help me solve this as well?

**Comment by tsutsuku** [comment]
@SM-26 https://github.com/tuya/tuya-home-assistant/discussions/81

**Comment by SM-26** [comment]
> @SM-26 #81

OMG thank you, yes!
I really missed that very last bit.

the username and password are the APP
not the iot.tuya.com

Finally got it working.

**Comment by tsutsuku** [comment]
@SM-26 We write more wiki, some new ones can read them.

**Comment by SM-26** [comment]
Yep, after re-reading the wiki, and updating to 1.4 , everything works like a charm.
Huge thanks.
